GPU/Vulkan/StencilBufferVulkan.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2014- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#include "base/logging.h"
+
+#include "ext/native/thin3d/thin3d.h"
+#include "ext/native/thin3d/VulkanRenderManager.h"
+#include "Core/Reporting.h"
+#include "GPU/Vulkan/FramebufferVulkan.h"
+#include "GPU/Vulkan/FragmentShaderGeneratorVulkan.h"
+#include "GPU/Vulkan/ShaderManagerVulkan.h"
+#include "GPU/Vulkan/TextureCacheVulkan.h"
+#include "GPU/Vulkan/VulkanUtil.h"
+
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
+
+struct StencilValueUB {
+	uint32_t u_stencilValue[4];
+};
+
+static const char *stencil_fs = R"(#version 400
+#extension GL_ARB_separate_shader_objects : enable
+#extension GL_ARB_shading_language_420pack : enable
+layout (binding = 0) uniform sampler2D tex;
+layout(push_constant) uniform params {
+  int u_stencilValue;
+};
+layout (location = 0) in vec2 v_texcoord0;
+layout (location = 0) out vec4 fragColor0;
+
+void main() {
+  vec4 index = texture(tex, v_texcoord0);
+	int indexBits = int(floor(index.a * 255.99)) & 0xFF;
+	if ((indexBits & u_stencilValue) == 0)
+		discard;
+  fragColor0 = index.aaaa;
```

Comment:
nit: Tabs and spaces living in intolerable harmony (also stencil_vs.)

-[Unknown]